### PR TITLE
Add serde [Rust]

### PR DESCRIPTION
**Pull request by lthiery**
This provides some serde attributes allowing the proto definitions to be leveraging when deserializing JSON from the API.

This should not affect current Rust proto bindings at all, but needs to be audited within that scope [as procedural macros are unhygenic](https://doc.rust-lang.org/reference/procedural-macros.html).

Of note is the deserializer for bs58 which tolerates null fields in the JSON. [A ticket against Blockchain API](https://app.clubhouse.io/hlm/story/8722/when-payer-is-null-populate-json-response-with-owner) has been filed which might allow for the deserializer to no longer tolerate null fields.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,196 @@
+mod origin {
```
Comment (JayKickliter) [inline src/lib.rs:1]: what's the reasoning behind having all these module blocks?

Comment (lthiery) [reply src/lib.rs:1]: good point. i think i started with a copy-paste example and just kept going. i should fix this

**Comment by JayKickliter** [comment]
I see you just pushed to this PR. I take it that it's still alive?

**Comment by lthiery** [comment]
> I see you just pushed to this PR. I take it that it's still alive?

Yea i still use this stuff in transaction analysis in Rust. I keep the branch alive for personal reasons, but I can close the PR and move this to a personal repo if @madninja prefers

**Review by uhrssophie — APPROVED** [review]

**Review by uhrssophie — APPROVED** [review]

**Comment by madninja** [comment]
Can we close this @lthiery ?

**Comment by lthiery** [comment]
> Can we close this @lthiery ?

Yessir, closed.

**Pull request closed without merging**